File: scala-finch/src/main/scala/org/openapitools/models/CodedErrorBaseAllOf.scala
package org.openapitools.models

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import org.openapitools._
import org.openapitools.models.AnyType

/**
 * 
 * @param result 
 * @param msg 
 * @param code A string that identifies the error. 
 */
case class CodedErrorBaseAllOf(result: Option[AnyType],
                msg: Option[AnyType],
                code: Option[String]
                )

object CodedErrorBaseAllOf {
    /**
     * Creates the codec for converting CodedErrorBaseAllOf from and to JSON.
     */
    implicit val decoder: Decoder[CodedErrorBaseAllOf] = deriveDecoder
    implicit val encoder: ObjectEncoder[CodedErrorBaseAllOf] = deriveEncoder
}

File: fsharp-functions/OpenAPI/src/model/CodedErrorBaseAllOf.fs
namespace OpenAPI.Model

open System
open System.Collections.Generic
open Newtonsoft.Json
open OpenAPI.Model.AnyType

module CodedErrorBaseAllOf = 

  //#region CodedErrorBaseAllOf

  [<CLIMutable>]
  type CodedErrorBaseAllOf = {
    [<JsonProperty(PropertyName = "result")>]
    Result : AnyType;
    [<JsonProperty(PropertyName = "msg")>]
    Msg : AnyType;
    [<JsonProperty(PropertyName = "code")>]
    Code : string;
  }
  
  //#endregion
  